x-pack/libbeat/common/aws/credentials.go

```diff
@@ -29,37 +29,53 @@ type ConfigAWS struct {
 }
 
 // GetAWSCredentials function gets aws credentials from the config.
-// If access_key_id and secret_access_key are given, then use them as credentials.
-// If role_arn is given, assume the IAM role instead.
-// If none of the above is given, then load from aws config file. If credential_profile_name is not
-// given, then load default profile from the aws config file.
+// If access keys given, use them as credentials.
+// If access keys are not given, then load from AWS config file. If credential_profile_name is not
+// given, default profile will be used.
+// If role_arn is given, assume the IAM role either with access keys or default profile.
 func GetAWSCredentials(config ConfigAWS) (awssdk.Config, error) {
-	logger := logp.NewLogger("get_aws_credentials")
-
 	// Check if accessKeyID or secretAccessKey or sessionToken is given from configuration
 	if config.AccessKeyID != "" || config.SecretAccessKey != "" || config.SessionToken != "" {
-		logger.Debug("Using access_key_id, secret_access_key and/or session_token for AWS credential")
-		awsConfig := defaults.Config()
-		awsCredentials := awssdk.Credentials{
-			AccessKeyID:     config.AccessKeyID,
-			SecretAccessKey: config.SecretAccessKey,
-		}
+		return getAccessKeys(config), nil
+	}
+	return getSharedCredentialProfile(config)
+}
 
-		if config.SessionToken != "" {
-			awsCredentials.SessionToken = config.SessionToken
-		}
+func getAccessKeys(config ConfigAWS) awssdk.Config {
+	logger := logp.NewLogger("getAccessKeys")
+	awsConfig := defaults.Config()
+	awsCredentials := awssdk.Credentials{
+		AccessKeyID:     config.AccessKeyID,
+		SecretAccessKey: config.SecretAccessKey,
+	}
 
-		awsConfig.Credentials = awssdk.StaticCredentialsProvider{
-			Value: awsCredentials,
-		}
-		return awsConfig, nil
+	if config.SessionToken != "" {
+		awsCredentials.SessionToken = config.SessionToken
+	}
+
+	awsConfig.Credentials = awssdk.StaticCredentialsProvider{
+		Value: awsCredentials,
 	}
 
+	// Set default region to make initial aws api call
+	awsConfig.Region = "us-east-1"
+
+	// Assume IAM role if iam_role config parameter is given
+	if config.RoleArn != "" {
+		logger.Debug("Using role arn and access keys for AWS credential")
+		return getRoleArn(config, awsConfig)
+	} else {
```

Comment:
Nit: no need for `else` as you are returning with `getRoleArn` above.

Reply:
Done, thanks for the review!